src/nvim/regexp.c

```diff
@@ -6984,10 +6984,14 @@ regprog_T *vim_regcomp(char_u *expr_arg, int re_flags)
   /*
    * First try the NFA engine, unless backtracking was requested.
    */
-  if (regexp_engine != BACKTRACKING_ENGINE)
-    prog = nfa_regengine.regcomp(expr, re_flags);
-  else
+  if (regexp_engine != BACKTRACKING_ENGINE) {
+    prog = nfa_regengine.regcomp(expr,
+                                 re_flags + (regexp_engine == AUTOMATIC_ENGINE
+                                             ? RE_AUTO
+                                             : 0));
```

Comment:
I don't really like it either. I think we should allow exceptions for cases like this, and allow 2-space indent instead. E.g.:

```c
  if (regexp_engine != BACKTRACKING_ENGINE) {
    prog = nfa_regengine.regcomp(expr,
      re_flags + (regexp_engine == AUTOMATIC_ENGINE ? RE_AUTO : 0));
```

Especially considering it saves 2 lines here, the vertical alignment thing seems unnecessary to me.